appinventor/components/src/com/google/appinventor/components/runtime/SpeechRecognizer.java

```diff
@@ -110,4 +127,59 @@ public void AfterGettingText(String result) {
     EventDispatcher.dispatchEvent(this, "AfterGettingText", result);
   }
 
+  /**
+   * Method from SpeechListener interface.
+   */
+  @Override
+  public void onPartialResult(String text) {
+    result = text;
+    AfterGettingText(result);
+  }
+
+  /**
+   * Method from SpeechListener interface.
+   */
+  @Override
+  public void onResult(String text) {
+    result = text;
+    AfterGettingText(result);
+    onDelete();
+  }
+
+  /**
+   * Method from SpeechListener interface.
+   */
+  @Override
+  public void onError(String message) {
+    result = message;
+    AfterGettingText(result);
+  }
+
+  @Override
+  public void onDelete() {
+    speechRecognizerController = null;
+    recognizerIntent = null;
+  }
+
+  @SimpleProperty(
+      category = PropertyCategory.BEHAVIOR,
+      description = "If set, an app can retain their older behaviour.")
+  public boolean UseLegacy() {
+      return useLegacy;
+  }
+
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_BOOLEAN,
+      defaultValue = "True")
+  @SimpleProperty(description = "Used to set UseLegacy property based on user's choice.")
+  public void UseLegacy(boolean useLegacy) {
+    this.useLegacy = useLegacy;
+  }
+
+  public void initialize(){
```

Comment:
Rather than use this method, you could simply have the UseLegacy setter instantiate the right object.

Reply:
Is setter called every time an app is run? 
If I move speechRecognizer instantiating code in the setter, won't it be instantiated only when the setter is called/used? 
A user need not always use the setter to set the value of UseLegacy, isn't it?

Reply:
The setter is only called if either the property in the designer is set to the non-default value or if the property is set from the blocks. You can call it from the constructor with the default value if you want to make sure that it is always called at least once.